docs/components/data_tables.md

```diff
@@ -0,0 +1,386 @@
+            header: Text('Header Text'),
+            rowsPerPage: 4,
+            columns: [
+              DataColumn(label: Text('Header A')),
+              DataColumn(label: Text('Header B')),
+              DataColumn(label: Text('Header C')),
+              DataColumn(label: Text('Header D')),
+            ],
+            source: _DataSource(context),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _Row {
+  _Row(
+    this.valueA,
+    this.valueB,
+    this.valueC,
+    this.valueD,
+  );
+
+  final String valueA;
+  final String valueB;
+  final String valueC;
+  final int valueD;
+
+  bool selected = false;
+}
+
+class _DataSource extends DataTableSource {
+  _DataSource(this.context) {
+    _rows = <_Row>[
+      _Row('Cell A1', 'CellB1', 'CellC1', 1),
+      _Row('Cell A2', 'CellB2', 'CellC2', 2),
+      _Row('Cell A3', 'CellB3', 'CellC3', 3),
+      _Row('Cell A4', 'CellB4', 'CellC4', 4),
+    ];
+  }
+
+  final BuildContext context;
+  List<_Row> _rows;
+
+  int _selectedCount = 0;
+
+  @override
+  DataRow getRow(int index) {
+    assert(index >= 0);
+    if (index >= _rows.length) return null;
+    final row = _rows[index];
+    return DataRow.byIndex(
+      index: index,
+      selected: row.selected,
+      onSelectChanged: (value) {
+        if (row.selected != value) {
+          _selectedCount += value ? 1 : -1;
+          assert(_selectedCount >= 0);
+          row.selected = value;
+          notifyListeners();
+        }
+      },
+      cells: [
+        DataCell(Text(row.valueA)),
+        DataCell(Text(row.valueB)),
+        DataCell(Text(row.valueC)),
+        DataCell(Text(row.valueD.toString())),
+      ],
+    );
+  }
+
+  @override
+  int get rowCount => _rows.length;
+
+  @override
+  bool get isRowCountApproximate => false;
+
+  @override
+  int get selectedRowCount => _selectedCount;
+}
+
+```
+
+### Key properties
+
+The following is an anatomy diagram of a data table:
+
+![Data table anatomy diagram](assets/data_tables/data-table-anatomy.png)
+
+A data table consists of the following properties:
+
+1. Container
+1. Column header names
+1. Sort button
+1. Row checkboxes
+1. Rows
+
+### Container
+
+| &nbsp; | Property |
+| --- | --- |
+| **Color** | style |
+| **Stroke color** | while setting `ThemeData` fill out `divideColor` parameter |
+
+### Column header names
+
+| &nbsp; | Property |
+| --- | --- |
+| **Label text** | Use `columns`, within `Column` set create `DataColumn` widget fill out `label` property with `Text` Widget|
+| **Typography** | Within the `Text` widget from note above fill out `style` property with `TextStyle` widget|
+| **Color** | Within the `Text` widget from note above fill out `style` property with `TextStyle` widget fill out the `color` property |
+
+### Sort button
+
+Sorting button can't be customized.
+
+### Row checkboxes
+
+| &nbsp; | Property |
+| --- | --- |
+| **active Color** | while setting `ThemeData` fill out `accentColor`|
+
+### Rows
+
```

Comment:
These should be level 4 headers because these are sub-headers of "Key properties"

Reply:
Done